docs/paths.adoc

```diff
@@ -0,0 +1,148 @@
+
+== Paths
+
+A `path` is a way to access a field or register. It is a compile-time list of
```

Comment:
suggestion: may be useful to point out the analogy with filesystem and url paths